lib/vsc/utils/mail.py

```diff
@@ -68,8 +67,45 @@ def __init__(self, mail_host=None, mail_to=None, mail_from=None, mail_subject=No
 class VscMail(object):
     """Class providing functionality to send out mail."""
 
-    def __init__(self, mail_host=None):
-        self.mail_host = mail_host
+    def __init__(self, mail_host=None, mail_port=None, smtp_auth_user=None, smtp_auth_password=None, smtp_use_starttls=False):
+
+        # we must not have the port in the host as this will fail the certificate check if needed
+        (host, port) = (mail_host, mail_port)
+        if mail_host:
+            try:
+                (host, port_) = mail_host.split(":")
+                port = port_ or port
+            except ValueError:
+                pass
+
+        self.mail_host = host
+        self.mail_port = port
+        self.smtp_auth_user = smtp_auth_user
+        self.smtp_auth_password = smtp_auth_password
+        self.smtp_use_starttls = smtp_use_starttls
+
+    def _connect(self):
+
+        logging.debug("Using %s as the mail host with authenticatio", self.mail_host)
+        if self.mail_host:
```

Comment:
use the defaults

Reply:
please use the defaults

Reply:
I went through the code again, defaults are used when there is nothing provided, so I am not sure I can follow here.

Reply:
i meant to replace the whole if/else with
```python
logging.debug("Using %s as the mail host and %d as mail port", self.mail_host, self.mail_port)
s = smtplib.SMTP(host=self.mail_host, port=self.mail_port)
```